File: 0551-student-attendance-record-i/0551-student-attendance-record-i.cpp
class Solution {
public:
    bool checkRecord(string s) {
        bool consecutive=false;
        bool award=false;
        int a=0;
        for(int i=0;i<s.size();i++){
            if(s[i]=='A'){
                a+=1;
            }
            if(s[i]=='L'){
                if(i+2<s.size() && i+1<s.size()){
                    if(s[i+1]=='L' && s[i+2]=='L'){
                        consecutive=true;
                        break;
                    }
                }
            }
            
        }
        if(a<2 && !consecutive){
                return true;
            }
            return false;
    }
};

File: 0551-student-attendance-record-i/0551-student-attendance-record-i.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        con=False
        a=0
        for i in range(len(s)):
            if s[i]=='A':
                a+=1
            if s[i]=='L':
                if(i+1<len(s) and i+2<=len(s)):
                    if(s[i+1]=='L' and s[i+2]=='L'):
                        con=True
                        break
        if(a<2 and not con):
            return True
        return False
